scripts/sceneNet_format.py

```diff
@@ -1,12 +1,69 @@
-import shutil
+import os
 import argparse
 from pathlib import Path
+from shutil import copyfile
+import numpy as np
+from PIL import Image 
 
-home = str(Path.home())
 parser = argparse.ArgumentParser(description='format')
-
-parser.add_argument('--dir', type=str, default= home + '/Downloads/train/0/', help='location where sceneNet dataset is extracted')
+parser.add_argument('--dir', type=str, default= '~/dataset/0/', help='location where sceneNet dataset is extracted from home directory')
+parser.add_argument('--keep',  action='store_true', help='Call to keep copy images instead of moving them')
 
 if __name__ == '__main__':
     args = parser.parse_args()
-    shutil.move(args.dir, "./vid2vid/datasets/sceneNet/")
\ No newline at end of file
+
+    home = str(Path.home())
+    in_dir = args.dir.replace('~', home)
+    out_dir = './vid2vid/datasets/sceneNet/'
+
+    assert os.path.isdir(in_dir), '%s is not a valid directory' % in_dir
+    assert os.path.isdir(out_dir), 'vid2vid repo not found within depth2room'
+
+    seq_num = 0
+
+    dir_list = os.listdir(in_dir)
+    dir_list.sort(key=lambda f: int(f.rsplit(os.path.extsep, 1)[0].rsplit(None,1)[-1]))
+    
+    for f in dir_list:
+        print('Processing sequence ' + str(seq_num) + '.')
+
+        in_dir_A = os.path.join(in_dir, f + '/depth')
+        in_dir_B = os.path.join(in_dir, f + '/photo')
+
+        out_dir_A = os.path.join(out_dir, 'train_A/seq' + str(seq_num).zfill(4))
+        out_dir_B = os.path.join(out_dir, 'train_B/seq' + str(seq_num).zfill(4))
+
+        if not os.path.exists(out_dir_A): 
+            os.makedirs(out_dir_A)
+        if not os.path.exists(out_dir_B): 
+            os.makedirs(out_dir_B)
+
+        for _, _, fnames in sorted(os.walk(in_dir_A)):
+            data_num = 0
+            fnames.sort(key=lambda f: int(f.rsplit(os.path.extsep, 1)[0].rsplit(None,1)[-1]))
+
+            for fname in fnames:
+                in_depth_name = os.path.join(in_dir_A, fname)
+                in_rgb_name = os.path.join(in_dir_B, fname.replace('png', 'jpg'))
+
+                out_depth_name = os.path.join(out_dir_A, str(seq_num).zfill(4) + '_' + str(data_num).zfill(4) + '.png')
+                out_rgb_name = os.path.join(out_dir_B, str(seq_num).zfill(4) + '_' + str(data_num).zfill(4) + '.jpg')
+
+                img_depth = Image.open(in_depth_name)
+                img_depth = np.array(img_depth, dtype=np.uint32)
+                img_depth = np.divide(img_depth, 100.0)
+                img_depth = img_depth.astype(np.uint8)
```

Comment:
What is the dynamic range of the images? wondering what made them choose uint32, we typically use uint16. Going to uint8, how much of detail do you lose in the image? also why divide by 100 and cast it to uint8?

Reply:
The depth images are describing distance in millimeters. I'm assuming they chose uint16 (correction) so they represent distance for far objects as well (65.5 m max.). Here I'm dividing by 100 to convert mm to cm. Also, since each sequence of the dataset is in indoor environments, it can be assumed that no object will be further than 5 meters.